<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DFindDPosDecimate</class>
 <widget class="QDialog" name="DFindDPosDecimate">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>645</width>
    <height>387</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dislocation position using the Burgers circuit construction</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>340</y>
     <width>191</width>
     <height>32</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
   <property name="centerButtons">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QCheckBox" name="shuffleIdx">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>270</y>
     <width>531</width>
     <height>22</height>
    </rect>
   </property>
   <property name="text">
    <string>Shuffle the order of atoms (Fisher-Yates algorithm)</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="interactive">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>250</y>
     <width>531</width>
     <height>22</height>
    </rect>
   </property>
   <property name="text">
    <string>Check this if you want to run interactive visualization of the calculation</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="ncalcdpos">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>290</y>
     <width>53</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
   <property name="value">
    <number>1</number>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>294</y>
     <width>241</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>- number of shuffles (calculations):</string>
   </property>
  </widget>
  <widget class="QLabel" name="hint">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>621</width>
     <height>251</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::NoFrame</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This calculation will find a boundary of a convex region that contains in its interior the intersection of the dislocation line with the (x,y) plane shown (this can be any plane in the crystal). This boundary will be obtained by successive bisections of the initial convex region by the line segments that represent all pairs of atoms in the block. We are now going to define this region that contains the dislocation by picking a set of atoms that represent the corners of its boundary. &lt;span style=&quot; color:#ff5500;&quot;&gt;These points have to be picked in a COUNTERCLOCKWISE direction and such that the polygon is CONVEX.&lt;/span&gt;&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;The relevant mouse events are:&lt;br/&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;click (left button) &lt;/span&gt;picks an atom&lt;br/&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;click (right button)&lt;/span&gt; removes the previously picked atom&lt;br/&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;double-click (left button)&lt;/span&gt; picks the last atom of the polygon and runs the calculation&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;Click [Cancel] below if you don't want to continue.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="textFormat">
    <enum>Qt::RichText</enum>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
   <property name="margin">
    <number>5</number>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DFindDPosDecimate</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DFindDPosDecimate</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
